transaction.go

```diff
@@ -7,6 +7,23 @@ import (
 	"github.com/lionelbarrow/braintree-go/nullable"
 )
 
+const (
+	TransactionStatusAuthorizationExpired   = "authorization_expired"
+	TransactionStatusAuthorizing            = "authorizing"
+	TransactionStatusAuthorized             = "authorized"
+	TransactionStatusGatewayRejected        = "gateway_rejected"
+	TransactionStatusFailed                 = "failed"
+	TransactionStatusProcessorDeclined      = "processor_declined"
+	TransactionStatusSettled                = "settled"
+	TransactionStatusSettlementConfirmed    = "settlement_confirmed"
+	TransactionStatusSettlementDeclined     = "settlement_declined"
+	TransactionStatusSettlementPending      = "settlement_pending"
+	TransactionStatusSettling               = "settling"
+	TransactionStatusSubmittedForSettlement = "submitted_for_settlement"
+	TransactionStatusVoided                 = "voided"
+	TransactionStatusUnrecognized           = "unrecognized"
```

Comment:
It looks like a nice pattern to use, I'm just curious how much breakage it'd cause for code that takes the value and stores it in a string field since go won't auto cast it to a string.